sourcegraph/learn
# Add sanitizer util function for SourcegraphSearch query input

### What should this PR do?
Resolves [DEVED-181](https://sourcegraph.atlassian.net/browse/DEVED-181) by adding a string sanitizer util function that can be used across the application to sanitize strings. Specifically, we can use it to address the zero-width space injection issue mentioned in #262 

### Why are we making this change?
- We want our `SourcegraphSearch` query examples to work and get users excited about search!

### What are the acceptance criteria? 
- Query examples should work across tutorials, including examples using different pattern types (literal, structural, regexp) & queries with keywords.
- Queries with extra spaces should have those spaces removed.

### How should this PR be tested?
- Check out the branch or take a look at the branch deploy PR and check that tutorial queries behave as expected on Sourcegraph.com.
- Add some extraneous spaces to queries in the search cell and check that they are removed as expected in the URL.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
